k9mail/src/main/java/com/fsck/k9/controller/ImapSyncInteractor.java

```diff
@@ -243,11 +226,30 @@ void syncRemoteDeletions(List<String> deletedMessageUids) throws IOException, Me
 
         if (moreMessages != null && moreMessages == MoreMessages.UNKNOWN) {
             final Date earliestDate = account.getEarliestPollDate();
-            int remoteStart = getRemoteStart(localFolder, imapFolder);
+            int remoteStart = syncHelper.getRemoteStart(localFolder, imapFolder);
             updateMoreMessages(earliestDate, remoteStart);
         }
     }
 
+    private void getAndOpenLocalFolder(SyncHelper syncHelper) throws MessagingException {
+        if (localFolder == null || !localFolder.isOpen()) {
```

Comment:
under what circumstances does this happen? if an instance of this class requires initialization before it can be used, and there are multiple entry points, maybe just use an `initalize()` method that takes care of the whole thing?

either way a method that starts with `get` should return something, not cause side effects like this.

(also see comment below about the `syncRemoteDeletions` control flow before proceeding on this)